code/core/src/main/java/com/adobe/marketing/mobile/internal/configuration/ConfigurationStateManager.kt

```diff
@@ -76,11 +76,8 @@ internal class ConfigurationStateManager {
     internal var environmentAwareConfiguration: Map<String, Any?> = mapOf()
         private set
 
-    /**
-     * Maintains a mapping between appId and the most recent download time when the config for that
-     * appId has been downloaded
-     */
-    private val configDownloadMap: MutableMap<String, Date> = mutableMapOf()
+    @VisibleForTesting
+    internal var lastDownloadedConfig: Pair<String, Date>? = null
```

Comment:
I think the current version uses a tiny optimization to cache the older configs and check their validity. I am aligned that this may not be that important to keep in favor of the proposed method. 
However, can we make the same changes in iOS too to make it consistant.

Reply:
Sure, if you agree this suggested approach won't break any current configuration workflows, I'll apply the same change on the iOS side as well.

Reply:
@yangyansong-adbe and I had a quick chat on the topic, posting here a quick summary for posterity. 

We took a look at the internal logic for Configuration, and in addition to the `configDownloadMap` the previously downloaded configs are cached locally. Since we are not clearing these configs on switch, we found that this logic can give us an advantage for situations where network connectivity is not available/poor, and the app logic switches between two app ids at runtime. 
For example in the case where we set appIdA (1) -> appIdB (2) -> appIdA (3), appId (3) could still recover the cached config if the timeout did not expire and it was successfully cached on step 1.